File: JobBoardService/src/main/java/com/project4/JobBoardService/DTO/InforUserDTO.java
package com.project4.JobBoardService.DTO;

import com.project4.JobBoardService.Entity.Permission;
import com.project4.JobBoardService.Entity.Role;
import com.project4.JobBoardService.Enum.Gender;
import lombok.Data;
import org.springframework.format.annotation.DateTimeFormat;

import java.time.LocalDate;
import java.util.List;
import java.util.Set;

@Data
public class InforUserDTO {
    private Long id;
    private String email;
    private String username;
    private String firstName;
    private String lastName;
    private Gender gender;
    private String imageUrl;
    private String bio;
    private Boolean isEnabled;
    private Set<Role> roles;
    private String numberphone;
    private String facebook;
    private List<CertificateDTO> certificates;
    private List<UserEducationDTO> userEducations;

    @DateTimeFormat(iso = DateTimeFormat.ISO.DATE)
    private LocalDate dateOfBirth;
    private String currentAddress;
}

File: jobboardmobile/lib/dto/Inforuser_dto.dart
import 'package:intl/intl.dart';

enum Gender { MALE, FEMALE, OTHER }

class InforUserDTO {
  int? id;
  String? email;
  String? username;
  String? firstName;
  String? lastName;
  Gender? gender;
  String? imageUrl;
  String? bio;
  String? numberphone;
  String? facebook;
  DateTime? dateOfBirth;
  String? currentAddress;

  InforUserDTO({
    this.id,
    this.email,
    this.username,
    this.firstName,
    this.lastName,
    this.gender,
    this.imageUrl,
    this.bio,
    this.numberphone,
    this.facebook,
    this.dateOfBirth,
    this.currentAddress,
  });

  factory InforUserDTO.fromJson(Map<String, dynamic> json) {
    return InforUserDTO(
      id: json['id'] as int?,
      email: json['email'] as String?,
      username: json['username'] as String?,
      firstName: json['firstName'] as String?,
      lastName: json['lastName'] as String?,
      gender: json['gender'] != null
          ? Gender.values
              .firstWhere((e) => e.toString().split('.').last == json['gender'])
          : null,
      imageUrl: json['imageUrl'] as String?,
      bio: json['bio'] as String?,
      numberphone: json['numberphone'] as String?,
      facebook: json['facebook'] as String?,
      dateOfBirth: json['dateOfBirth'] != null
          ? DateTime.parse(json['dateOfBirth'])
          : null,
      currentAddress: json['currentAddress'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    final DateFormat formatter = DateFormat('yyyy-MM-dd');
    return {
      'id': id,
      'email': email,
      'username': username,
      'firstName': firstName,
      'lastName': lastName,
      'gender': gender?.toString().split('.').last,
      'imageUrl': imageUrl,
      'bio': bio,
      'numberphone': numberphone,
      'facebook': facebook,
      'dateOfBirth':
          dateOfBirth != null ? formatter.format(dateOfBirth!) : null,
      'currentAddress': currentAddress,
    };
  }
}
